mage_ai/ai/llm_pipeline_wizard.py

```diff
@@ -247,6 +257,42 @@ async def async_generate_pipeline_with_description(self, pipeline_description: s
         await asyncio.gather(*block_tasks)
         return blocks
 
+    def __insert_comments_in_functions(self, code: str, function_comments_json: dict):
+        # Parse the input code into an abstract syntax tree (AST).
+        tree = ast.parse(code)
+        # Traverse the AST and find function definitions.
+        for node in ast.walk(tree):
+            if isinstance(node, ast.FunctionDef):
+                function_name = node.name
+                if function_comments_json.get(function_name):
+                    comment_text = function_comments_json[function_name]
+                    # Insert a comment node below a given node.
+                    if isinstance(node.body[0], ast.Expr) and \
+                       isinstance(node.body[0].value, ast.Constant):
+                        # If there is existing doc string, combine the new comment with it.
+                        existing_comment_node = node.body[0]
+                        existing_comment_text = node.body[0].value.value
+                        new_comment = ast.Expr(
+                            value=ast.Str(s=f"{comment_text}\n{existing_comment_text}"))
+                        node.body.remove(existing_comment_node)
+                    else:
+                        # Add newly generated doc string.
+                        new_comment = ast.Expr(value=ast.Str(s=comment_text))
+                    node.body.insert(0, new_comment)
+        return ast.unparse(tree)
+
+    async def async_generate_comment_for_block(self, block_content: str) -> str:
+        prompt_template = PromptTemplate(
+            input_variables=[
+                'block_content',
+            ],
+            template=PROMPT_FOR_FUNCTION_COMMENT,
+        )
+        chain = LLMChain(llm=self.llm, prompt=prompt_template)
+        function_comments = await chain.arun(block_content=block_content)
+        function_comments_json = json.loads(function_comments)
+        return self.__insert_comments_in_functions(block_content, function_comments_json)
```

Comment:
i think the variable names here should be flipped. function_comments here is the json string, and function_comments_json is the object